File: Valid Palindrome (125)/125.java
// Runtime: 6 ms
// Beats 57.43% of Java submissions

class Solution {
    public boolean isPalindrome(String s) {
        char[] stringArr = s.toLowerCase().toCharArray();
        int beginning = 0;
        int end = stringArr.length - 1;
        
        while (beginning < end) {
            if (!Character.isLetterOrDigit(stringArr[beginning])) {
                beginning++;
            }
            else if (!Character.isLetterOrDigit(stringArr[end])) {
                end--;
            } else {
                if (stringArr[beginning] != stringArr[end]) {
                    return false;
                }
                beginning++;
                end--;   
            }
        }
        
        return true;
    }
}

File: Valid Palindrome (125)/125.py
# Runtime: 60 ms
# Beats 84.04% of Python submissions

class Solution:
    def isPalindrome(self, s):
        """
        :type s: str
        :rtype: bool
        """
        beginning = 0
        end = len(s) - 1
        s = s.lower()
        
        while beginning < end:
            if not s[beginning].isalnum():
                beginning += 1
            elif not s[end].isalnum():
                end -= 1
            else:
                if s[beginning] != s[end]:
                    return False
                beginning += 1
                end -= 1
        
        return True
        